src/redux/books/books.js

```diff
@@ -0,0 +1,25 @@
+// Actions
+const ADD_BOOK = 'ADD_BOOK';
+const REMOVE_BOOK = 'REMOVE_BOOK';
+
+// Action Creators
+export const addBook = (book) => ({ type: ADD_BOOK, payload: book });
+export const removeBook = (id) => ({ type: REMOVE_BOOK, payload: { id } });
+
+// Reducer
+const books = (state = [], action) => {
+  switch (action.type) {
+    // do reducer stuff
+    case ADD_BOOK:
+      return [...state, {
+        id: state.length,
+        name: action.payload.name,
+        author: action.payload.author,
+      }];
```

Comment:
- [ ] There is no need to hardcode these properties here. They are already accessible as `action.payload` as defined in your `addBook` action creator. Instead we can have something like so:

  ```
  case ADD_BOOK:
    return [
      ...state,
      action.payload
    ];
  ```